actors/market/src/lib.rs

```diff
@@ -266,16 +246,23 @@ impl Actor {
         let baseline_power = request_current_baseline_power(rt)?;
         let (network_raw_power, _) = request_current_network_power(rt)?;
 
-        // Drop invalid deals
+        struct ValidDeal {
+            proposal: DealProposal,
+            cid: Cid,
+            allocation: AllocationID,
+        }
+
+        // Deals that passed validation.
+        let mut valid_deals: Vec<ValidDeal> = Vec::with_capacity(params.deals.len());
+        // CIDs of valid proposals.
         let mut proposal_cid_lookup = BTreeSet::new();
-        let mut valid_proposal_cids = Vec::new();
-        let mut valid_deals = Vec::with_capacity(params.deals.len());
         let mut total_client_lockup: BTreeMap<ActorID, TokenAmount> = BTreeMap::new();
         let mut total_provider_lockup = TokenAmount::zero();
 
         let mut valid_input_bf = BitField::default();
```

Comment:
Probably not something we want to get into right now BUT it would be strictly better to use the batch info return value instead of a bitfield for valid_deals in the return.  The motivation for making batch info came from friction users have with psd's lack of error reporting. 
 Breaking return values for a method number in place shouldn't be nearly as bad as breaking params since nodes can always handle this.  But this is a method with the potential for use across many ecosystem tools so could see wanting to wait for a FIP.

Reply:
I agree but don't want to change this return value schema in nv17